src/lib/core/components/visualizations/implementations/BarplotVisualization.tsx

```diff
@@ -48,16 +50,28 @@ function FullscreenComponent(props: VisualizationProps) {
 }
 
 function createDefaultConfig(): BarplotConfig {
-  return {};
+  return {
+    dependentAxisLogScale: false,
+    valueSpec: 'count',
+  };
 }
 
+type ValueSpec = t.TypeOf<typeof ValueSpec>;
+const ValueSpec = t.keyof({ count: null, proportion: null });
+
 type BarplotConfig = t.TypeOf<typeof BarplotConfig>;
 // eslint-disable-next-line @typescript-eslint/no-redeclare
-const BarplotConfig = t.partial({
-  xAxisVariable: VariableDescriptor,
-  overlayVariable: VariableDescriptor,
-  facetVariable: VariableDescriptor,
-});
+const BarplotConfig = t.intersection([
+  t.type({
+    dependentAxisLogScale: t.boolean,
+    valueSpec: ValueSpec,
+  }),
+  t.partial({
+    xAxisVariable: VariableDescriptor,
```

Comment:
Is `xAxisVariable` really optional?

Reply:
Yes. When you first open up a new Barplot - there aren't any variables selected by the user.

Reply:
Oh yeah... :upside_down_face: